packages/schedule/src/Schedule.js

```diff
@@ -4,323 +4,390 @@
-            'polyfill in older browsers. https://fb.me/react-polyfills',
-        );
+var deadlineObject = {
+  timeRemaining,
+  didTimeout: false,
+};
+
+function ensureHostCallbackIsScheduled() {
+  if (isPerformingWork) {
+    // Don't schedule work yet; wait until the next time we yield.
+    return;
+  }
+  // Schedule the host callback using the earliest timeout in the list.
+  var timesOutAt = firstCallbackNode.timesOutAt;
+  if (!isHostCallbackScheduled) {
+    isHostCallbackScheduled = true;
+  } else {
+    // Cancel the existing host callback.
+    cancelCallback();
+  }
+  requestCallback(flushWork, timesOutAt);
+}
+
+function flushFirstCallback(node) {
+  var flushedNode = firstCallbackNode;
+
+  // Remove the node from the list before calling the callback. That way the
+  // list is in a consistent state even if the callback throws.
+  var next = firstCallbackNode.next;
+  if (firstCallbackNode === next) {
+    // This is the last callback in the list.
+    firstCallbackNode = null;
+    next = null;
+  } else {
+    var previous = firstCallbackNode.previous;
+    firstCallbackNode = previous.next = next;
+    next.previous = previous;
+  }
+
+  flushedNode.next = flushedNode.previous = null;
+
+  // Now it's safe to call the callback.
+  var callback = flushedNode.callback;
+  callback(deadlineObject);
+}
+
+function flushWork(didTimeout) {
+  isPerformingWork = true;
+  deadlineObject.didTimeout = didTimeout;
+  try {
+    if (didTimeout) {
+      // Flush all the timed out callbacks without yielding.
+      while (firstCallbackNode !== null) {
+        // Read the current time. Flush all the callbacks that expire at or
+        // earlier than that time. Then read the current time again and repeat.
+        // This optimizes for as few performance.now calls as possible.
+        var currentTime = getCurrentTime();
+        if (firstCallbackNode.timesOutAt <= currentTime) {
+          do {
+            flushFirstCallback();
+          } while (
+            firstCallbackNode !== null &&
+            firstCallbackNode.timesOutAt <= currentTime
+          );
+          continue;
+        }
+        break;
       }
-      if (typeof localCancelAnimationFrame !== 'function') {
-        console.error(
-          "This browser doesn't support cancelAnimationFrame. " +
-            'Make sure that you load a ' +
-            'polyfill in older browsers. https://fb.me/react-polyfills',
+    } else {
+      // Keep flushing callbacks until we run out of time in the frame.
+      if (firstCallbackNode !== null) {
+        do {
+          flushFirstCallback();
+        } while (
+          firstCallbackNode !== null &&
+          getFrameDeadline() - getCurrentTime() > 0
         );
       }
     }
+  } finally {
+    isPerformingWork = false;
+    if (firstCallbackNode !== null) {
+      // There's still work remaining. Request another callback.
+      ensureHostCallbackIsScheduled(firstCallbackNode);
```

Comment:
@acdlite Seems `ensureHostCallbackIsScheduled` doesn't use any argument in the function body, can I ask that why we pass the `firstCallbackNode` here?

Reply:
For Flow I think

Reply:
hmm wait this doesn't make sense. I dunno.

Reply:
I think in the initial commit (7c4ff13) `ensureHostCallbackIsScheduled ` expected a param, but the second commit (5f74f04) changed it to just read from the package-level `firstCallbackNode` variable since that's always the value that was passed in anyway. Seems like the call sites were just overlooked as part of this change.

Reply:
I think the lint should warn on this but don't know why not